Improvements to eval_forward, getgamma, linear solvers

leespen1 opened the pull request:
Changed getgamma to have specified input types, an exception for invalid order provided, and a warning for invalid stages provided.

Added warning for maximum number of Jacobi iterations used.

Added "built-in" linear solver type.

Changed eval_forward to evolve an array of multiple initial state vectors, added arguments for more control over how much of the state evolution is saved, and changed several positional arguments to keyword arguments.

andersp reviewed (changes requested):
The getgamma() routine needs to be redone. It only needs a couple of more if-clauses for the newly implemented values of 'order'
In the linear_solvers struct, a more descriptive keyword for the "built-in" solver would be "direct" or "Gausian elimination"

leespen1 commented:
Simplified getgamma, and also changed behavior to default to the highest-stage variant of a method when the number of stages given is invalid (simpler than finding closest number of stages).

Also renamed the "built-in" linear solver functions to be more descriptive ("gaussian elimination").
